pyiceberg/io/pyarrow.py

```diff
@@ -1594,29 +1595,88 @@ def parquet_path_to_id_mapping(
     return result
 
 
-def fill_parquet_file_metadata(
-    data_file: DataFile,
+@dataclass
+class DataFileStatistics:
+    record_count: int
+    column_sizes: Dict[int, int]
+    value_counts: Dict[int, int]
+    null_value_counts: Dict[int, int]
+    nan_value_counts: Dict[int, int]
+    column_aggregates: Dict[int, StatsAggregator]
+    split_offsets: Optional[List[int]] = None
```

Comment:
Any specific reason that the `split_offsets` is nullable in contrast to the rest.

Reply:
I just carried its type hint over from the existing DataFile class: https://github.com/apache/iceberg-python/blob/d3db8401cd006fcbacf1d9a2502248ce4228ff06/pyiceberg/manifest.py#L363

Should it not be optional?

Reply:
Just decided to  updated it based on the code